Automattic/wp-calypso
# Launchpad: Clean up task-helper  (Apply Reverted changes)

### Proposed change
* Update the e2e tests to be more flexible with the checklist item. The items can be links or buttons.
* Restore the reverted changes applied by the #86960 PR and reverted by the PR #87204


### Context
It previously failed 2 times because the Pre-Release E2E Tests suite only run when changes are introduced on the trunk by default only the test suite from the trunk brunch and if we try to run the manually test suite on teamcity it by default uses the trunk branch and the wpcalypso env.

## Testing instructions
* Go to the [TeamCity pipeline](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Calypso_E2E_Pre_Release?branch=revert-87262-revert-87218-revert-87204-revert-86960-update%2Frefactor-task-helper&mode=builds#all-projects). You need to log in on the team city site.

* Click on the 3 dots button
![image](https://github.com/Automattic/wp-calypso/assets/38718/9d25487b-2079-4aa0-bc04-bc13e5105a7b)
* Go to the changes tab and set the commit ed3238dd5cba8bc96bcd202cfe7dc44702ce5654

![image](https://github.com/Automattic/wp-calypso/assets/38718/aa87c333-d94a-4f21-8580-d6000986f594)
* Go to the parameter tab and set the value with calypso.live URL generated on this PR (this moment https://container-compassionate-vaughan.calypso.live/)
![image](https://github.com/Automattic/wp-calypso/assets/38718/57afda4b-689b-4b54-8e92-7d743390df24)
* Run the tests and check if the pipe failed. It will reproduce the failed scenario that caused the revert. 
* Run the tests again but using the hash from the last commit available on this PR
![image](https://github.com/Automattic/wp-calypso/assets/38718/6c5c6c99-7e63-4a36-be04-528b4ee10d78)
* Check if the tests are passing, it confirm the last change fixed the pipeline and the frontend code is running well with the suite


### Context
During the deployment, we had the error described above on our `Pre-Release E2E Tests` task. A few minutes later we had issues with the CI servers as described here ->> p1707172064283299-slack-C02DQP0FP  as you can see [here](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Calypso_E2E_Pre_Release?branch=revert-87204-revert-86960-update%2Frefactor-task-helper&buildTypeTab=overview&mode=builds) 

After the revert, I rerun the same test against this branch multiple and all tests are passing consistently. 

The reverted changes were related to the launchpad and the test failed steps before we rendered the launchpad, so I am comfortable to suggest trying to merge again the PR.  



### Original Error
----
``` 
======= Failed test run #1 ==========
  locator.click: Timeout 10000ms exceeded.
  =========================== logs ===========================
  waiting for getByRole('button', { name: 'Name your blog' })
  ============================================================
  at Object.click (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/onboarding/signup__start-writing-tailored.ts:50:64)
      at Promise.then.completed (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:298:28)
      at callAsyncCircusFn (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:231:10)
      at _callCircusTest (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:316:40)
      at _runTest (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:252:3)
      at _runTestsForDescribeBlock (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:126:9)
      at _runTestsForDescribeBlock (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:121:9)
      at run (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:71:3)
      at runAndTransformResultsToJestFormat (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
      at jestAdapter (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
      at runTestInternal (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:367:16)
      at runTest (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:444:34)
      at Object.worker (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/testWorker.js:106:12)
======= Failed test run #2 ==========
  locator.click: Timeout 10000ms exceeded.
  =========================== logs ===========================
  waiting for getByRole('button', { name: 'Name your blog' })
  ============================================================
  at Object.click (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/onboarding/signup__start-writing-tailored.ts:50:64)
      at Promise.then.completed (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:298:28)
      at callAsyncCircusFn (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/utils.js:231:10)
      at _callCircusTest (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:316:40)
      at _runTest (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:252:3)
      at _runTestsForDescribeBlock (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:126:9)
      at _runTestsForDescribeBlock (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:121:9)
      at run (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/run.js:71:3)
      at runAndTransformResultsToJestFormat (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
      at jestAdapter (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
      at runTestInternal (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:367:16)
      at runTest (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/runTest.js:444:34)
      at Object.worker (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/jest-runner/build/testWorker.js:106:12)
      ```
